### PR TITLE
Add github workflow to test non-zero shard/realm 

### LINKED ISSUE #11540: Create Github Action to run tests with non-zero shard and realm
### Problem

Need to run periodic tests with non-zero shard and realm

### Solution

- Have a scheduled workflow to test main branch with non-zero shard and realm
- Allow selecting a different branch for manual execution

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds a workflow to test non-zero shard/realm

- Add github workflow to test non-zero shard/realm
- Don't fail-fast in jobs with matrix

**Related issue(s)**:

Fixes #11540 

**Notes for reviewer**:

There are tests failing in web3: https://github.com/hiero-ledger/hiero-mirror-node/actions/runs/16380493136?pr=11608

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/nzr-test.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          - grpc
+          - importer
+          - monitor
+          - rest
+          - rest-java
+          - rosetta
+          - web3
+        schema:
+          - v1
+          - v2
```
Comment (steven-sheehy) [inline .github/workflows/nzr-test.yml]: I don't think we need to test v2 with NZR since if it works with v1 it should with v2 since it's the same code. Can remove.

File: .github/workflows/nzr-test.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          MIRROR_NODE_SCHEMA: ${{ matrix.schema}}
+          SPRING_PROFILES_ACTIVE: ${{ matrix.schema}}
+        run: ./gradlew :${{matrix.project}}:build --scan ${{ secrets.GRADLE_ARGS }}
+
+      - name: Build and test web3 with modularized code
+        env:
+          HIERO_MIRROR_WEB3_EVM_MODULARIZEDSERVICES: "true"
+          HIERO_MIRROR_WEB3_EVM_MODULARIZEDTRAFFICPERCENT: "1.0"
+        if: ${{ matrix.project == 'web3' && matrix.schema == 'v1'}}
+        run: ./gradlew :${{ matrix.project }}:build
```
Comment (steven-sheehy) [inline .github/workflows/nzr-test.yml]: I think we can just only test NZR with modularized and remove this step.

File: .github/workflows/nzr-test.yml
Comment (steven-sheehy) [inline .github/workflows/nzr-test.yml:1]: Prefer just `non-zero-realm.yml`

File: .github/workflows/nzr-test.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: "Test non-zero shard and realm network"
```
Comment (steven-sheehy) [inline .github/workflows/nzr-test.yml]:
```suggestion
name: "Non-Zero Realm"
```

File: .github/workflows/nzr-test.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    name: Build
+    env:
+      HIERO_MIRROR_COMMON_REALM: 1
+      HIERO_MIRROR_COMMON_SHARD: 1
+    runs-on: hiero-mirror-node-linux-large
+    strategy:
+      fail-fast: false
+      matrix:
+        project:
+          - common
```
Comment (steven-sheehy) [inline .github/workflows/nzr-test.yml]: Should we simplify it to be a single threaded test without a matrix by using `org.gradle.parallel=false`?

**Review by steven-sheehy — COMMENTED** [review]
Is it possible to quickly fix any of the test failures in this PR or do we need a separate ticket?

File: .github/workflows/nzr-test.yml (hunk 4 above)
Comment (xin-hedera) [reply .github/workflows/nzr-test.yml]: that may take too long. not a big deal for scheduled runs, however it may be frustrating for runs triggered for a specific branch.

File: .github/workflows/non-zero-realm.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        uses: step-security/harden-runner@6c439dc8bdf85cadbbce9ed30d1c7b959517bc49 # v2.12.2
+        with:
+          egress-policy: audit
+
+      - name: Checkout Code
+        if: ${{ github.event_name != 'workflow_dispatch' }}
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Checkout Code from Branch
+        if: ${{ github.event_name == 'workflow_dispatch' }}
```
Comment (steven-sheehy) [inline .github/workflows/non-zero-realm.yml]: Is it possible to combine and have a conditional inside the `with.ref`?

File: .github/workflows/non-zero-realm.yml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        uses: actions/setup-java@c5195efecf7bdfc987ee8bae7a71cb8b11521c00 # v4.7.1
+        with:
+          distribution: temurin
+          java-version: 21
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@ac638b010cf58a27ee6c972d7336334ccaf61c96 # v4
+
+      - name: Execute Gradle
+        run: ./gradlew :${{matrix.project}}:build --scan ${{ secrets.GRADLE_ARGS }}
```
Comment (steven-sheehy) [inline .github/workflows/non-zero-realm.yml]: We can probably omit the build scan for this.

```suggestion
        run: ./gradlew :${{matrix.project}}:build
```

File: .github/workflows/non-zero-realm.yml (hunk 5 above)
Comment (xin-hedera) [reply .github/workflows/non-zero-realm.yml]: possible, let me confirm if the logic in the checkout action uses `github.ref` by default.

**Review by steven-sheehy — APPROVED** [review]
LGTM
